apps/hash-graph/lib/graph/src/store/query/filter.rs

```diff
@@ -518,6 +692,169 @@ mod tests {
         );
     }
 
+    #[test]
+    fn for_property_type_constrains_values_on() {
+        let url = DataTypeVertexId {
+            base_id: BaseUrl::new(
+                "https://blockprotocol.org/@blockprotocol/types/data-type/text/".to_owned(),
+            )
+            .expect("invalid base url"),
+            revision_id: OntologyTypeVersion::new(1),
+        };
+
+        let expected = json! {{
+          "all": [
+            { "equal": [
+              { "path": ["dataTypes", "*", "baseUrl"] },
+              { "parameter": url.base_id }
+            ]},
+            { "equal": [
+              { "path": ["dataTypes", "*", "version"] },
+              { "parameter": url.revision_id }
+            ]}
+          ]
+        }};
+
+        test_filter_representation(
+            &Filter::<PropertyTypeWithMetadata>::for_ontology_edge_by_data_type_vertex_id(
+                &url,
+                OntologyEdgeKind::ConstrainsValuesOn,
+            ),
+            &expected,
+        );
+    }
+
+    #[test]
+    fn for_property_type_constrains_properties_on() {
+        let url = PropertyTypeVertexId {
+            base_id: BaseUrl::new(
+                "https://blockprotocol.org/@blockprotocol/types/property-type/text/".to_owned(),
+            )
+            .expect("invalid base url"),
+            revision_id: OntologyTypeVersion::new(1),
+        };
+
+        let expected = json! {{
```

Comment:
small nit: does `rustfmt` add this space after `json!`?

Reply:
Yes, it does, and it's annoying. I replaced it with `json!({` instead: [Replace json! braces](https://github.com/hashintel/hash/pull/2208/commits/c522386388c9a130d9b2ac27b508709456198b91)